common/lib/src/realm_types.dart

```diff
@@ -227,3 +228,248 @@ class RealmValue {
   @override
   String toString() => 'RealmValue($value)';
 }
+
+/// A base type for the supported geospatial shapes.
+sealed class GeoShape {}
+
+/// A point on the earth's surface.
+///
+/// It cannot be persisted as a property on a realm object.
+///
+/// Instead, you must use a custom embedded object with the following structure:
+/// ```dart
+/// @RealmModel(ObjectType.embeddedObject)
+/// class _Location {
+///   final String type = 'Point';
+///   final List<double> coordinates = const [0, 0];
+///
+///   // The rest of the class is just convenience methods
+///   double get lon => coordinates[0];
+///   set lon(double value) => coordinates[0] = value;
+///
+///   double get lat => coordinates[1];
+///   set lat(double value) => coordinates[1] = value;
+///
+///   GeoPoint toGeoPoint() => GeoPoint(lon: lon, lat: lat);
+/// }
```

Comment:
I don't think we should include realm models in the SDK code itself. 

There is a phase 2 to this project, which will introduce persist-able classes. I'm following the other SDKs here and not limiting our freedom too much.

The idea about generating it in the generator and map to GeoPoint I like, but it may conflict with the realm-core design of phase 2.